lib/core/versions/latest/CreateOperation.ts

```diff
@@ -85,10 +97,58 @@ export default class CreateOperation implements OperationModel {
   public static async parse (operationBuffer: Buffer): Promise<CreateOperation> {
     const operationJsonString = operationBuffer.toString();
     const operationObject = await JsonAsync.parse(operationJsonString);
-    const createOperation = await CreateOperation.parseObject(operationObject, operationBuffer, false);
+    let createOperation;
+    if (typeof operationObject.suffix_data === 'string') {
+      createOperation = await CreateOperation.parseObject(operationObject, operationBuffer, false);
+    } else {
+      createOperation = CreateOperation.parseJcsObject(operationObject, operationBuffer);
+    }
     return createOperation;
   }
 
+  /**
+   * Parse the given operation object as a CreateOperation
+   * @param operationObject The operationObject is a json object with no encoding
+   * @param operationBuffer The buffer format of the operationObject
+   */
+  public static parseJcsObject (operationObject: any, operationBuffer: Buffer): CreateOperation {
+    let expectedPropertyCount = 3;
+    const properties = Object.keys(operationObject);
+    if (properties.length !== expectedPropertyCount) {
+      throw new SidetreeError(ErrorCode.CreateOperationMissingOrUnknownProperty);
+    }
+
+    if (operationObject.type !== OperationType.Create) {
+      throw new SidetreeError(ErrorCode.CreateOperationTypeIncorrect);
+    }
+
+    CreateOperation.validateSuffixData(operationObject.suffix_data);
+    const suffixData = {
+      deltaHash: operationObject.suffix_data.delta_hash,
+      recoveryCommitment: operationObject.suffix_data.recovery_commitment
+    };
+
+    // For compatibility with data pruning, we have to assume that `delta` may be unavailable,
+    // thus an operation with invalid `delta` needs to be processed as an operation with unavailable `delta`,
+    // so here we let `delta` be `undefined`.
+    let delta;
```

Comment:
Sanity: you are intentionally leaving out the `anchorFileMode` portion of the checks for later correct?

Reply:
Yes. Now it's just long form so I thought we could get this change in later. Do you think we should have it here? It's not a lot of work to add it tbh.